anomdetect/src/main/java/com/expedia/adaptivealerting/anomdetect/AnomalyDetectorManager.java

```diff
@@ -107,8 +107,9 @@ public AnomalyDetector detectorFor(MappedMpoint mappedMpoint) {
     public MappedMpoint classify(MappedMpoint mappedMpoint) {
         notNull(mappedMpoint, "mappedMpoint can't be null");
         final AnomalyDetector detector = detectorFor(mappedMpoint);
+        MappedMpoint newMappedMpoint = detector == null ? null : detector.classify(mappedMpoint);
```

Comment:
1) Don't pass null detector into MonitoredDetector. If the detector is null just return null from the method.
2) We don't want to create a new wrapper with every classification. The wrapper is what we store in the map.